### ISSUE TITLE
Timestamp microservice:unexpected Behavior in API Endpoint Handling Empty Date Parameters

### ISSUE THREAD

**Issue by Ajay-2005** [freeCodeCamp/freeCodeCamp#51891]
### Describe the Issue

The API endpoint handling empty date parameters is not functioning as expected. According to the test cases, when the API receives an empty date parameter, it should return the current time in a JSON object with both a unix key (representing the Unix timestamp of the current time) and a utc key (representing the current time in the UTC format). However, the current implementation is not returning the expected output for empty date parameters.

### Affected Page

https://www.freecodecamp.org/learn/back-end-development-and-apis/back-end-development-and-apis-projects/timestamp-microservice

### Your code

// index.js
// where your node app starts

// init project
var express = require('express');
var app = express();

// enable CORS (https://en.wikipedia.org/wiki/Cross-origin_resource_sharing)
// so that your API is remotely testable by FCC 
var cors = require('cors');
app.use(cors({optionsSuccessStatus: 200}));  // some legacy browsers choke on 204

// http://expressjs.com/en/starter/static-files.html
app.use(express.static('public'));

// http://expressjs.com/en/starter/basic-routing.html
app.get("/", function (req, res) {
  res.sendFile(__dirname + '/views/index.html');
});

// your first API endpoint... 
app.get("/api/hello", function (req, res) {
  res.json({greeting: 'hello API'});
});

app.get("/api", function (req, res) {
  const date = new Date(); // Create a new Date object
  res.json({
    "unix": date.valueOf(),
    "utc": date.toUTCString()
  });
});

app.get("/api/:date", function (req, res) {
  const input = req.params.date;
  let date;

  if (!isNaN(input) && input > 0) {
    date = new Date(parseInt(input));
  } else {
    date = new Date(input); 
  }
  if (!isNaN(date)) {
    res.json({
      "unix": date.valueOf(),
      "utc": date.toUTCString()
    });
  } else {
    res.json({ error: "Invalid Date" });
  }
});

function isValidDate(dateString) {
  const date = new Date(dateString);
  return !isNaN(date);
}

// listen for requests :)
var listener = app.listen(process.env.PORT, function () {
  console.log('Your app is listening on port ' + listener.address().port);
}); 

### Expected behavior

Passed:An empty date parameter should return the current time in a JSON object with a unix key
Passed:An empty date parameter should return the current time in a JSON object with a utc key

### Screenshots

https://github.com/freeCodeCamp/freeCodeCamp/assets/125485357/f15599b4-fb39-4077-8316-f104f3b7490a

### System

- OS: windows 11
- Browser: chrome
- Version: 117.0.5938.149

**Comment by vishal-sharma-369** [comment]
I can work on this issue. Please assign me this issue

**Comment by Ajay-2005** [comment]
> In which file are you referring to??
Hi
The issue with the back-end developer certification in the timestamp microservice service project is that it always fails the test cases. Even though the code is correct, I believe the problem lies in defining the test cases. 

The issue is that these two test cases always fail:

1.An empty date parameter should return the current time in a JSON object with a unix key.
2.An empty date parameter should return the current time in a JSON object with a utc key.

**Comment by Ajay-2005** [comment]
Hi @vishal-sharma-369 

The problem with the back-end developer certification lies in the timestamp microservice service project, where it consistently fails the test cases. Even though the code is correct, I believe the issue stems from defining the test cases.

The issue is that these two test cases always fail:

1.An empty date parameter should return the current time in a JSON object with a unix key.
2.An empty date parameter should return the current time in a JSON object with a utc key.

**Comment by vishal-sharma-369** [comment]
@Ajay-2005 Yes, the code is right. The issue might be in defining test cases.
